### DISCUSSION TITLE
Codeblock edit mode

### DISCUSSION THREAD

**Discussion by char-mainewang** [twilio-labs/paste#3134] [General] [answered]
Hey Paste team,

I'm in the process of recreating my designs from Evergreen to Paste. We have a few places where customers can write SQL queries (i.e. Reverse ETL Model, SQL Computed traits). For Reverse ETL, customers currently have a view and edit mode, i'm currently thinking of replacing the view mode with the codeblock component but I was wondering if the codeblock has an edit mode?

View mode on the model overview page
<img width="1687" alt="image" src="https://user-images.githubusercontent.com/87334135/228658087-b71b4af6-ddb7-4085-a4f5-82ef5889b098.png">

Edit mode in the query builder page
<img width="1329" alt="image" src="https://user-images.githubusercontent.com/87334135/228658168-0a5083e9-19d1-484f-a2c4-917e7812cab5.png">

**Comment by nkrantz** [comment, marked as answer]
Hi @char-mainewang, we do have a Code Editor component in our roadmap backlog, but nothing currently in the system. The best bet for now will be to build a custom component. We'd recommend using the [Monaco Code Editor](https://microsoft.github.io/monaco-editor/), as it's a reliable and widely used option.

If you could also fill out [a dependency request](https://airtable.com/shrvMvL3BG6Zm0f9l) for the Code Editor, that would be great!
